### PR TITLE
fix CID 1373512 and CID 1373506 in nro

### PR DESCRIPTION
**Pull request by h4ng3r**

### CODE REVIEW

File: libr/bin/p/bin_nro.c [hunk 1]
```diff
@@ -215,24 +215,33 @@ static void walkSymbols (RBinFile *bf, RBinNROObj *bin, ut64 symtab, ut64 strtab
 			ut64 pltSym = readLE64 (bf->buf, relplt + (import * 24));
 			RBinImport *imp = R_NEW0 (RBinImport);
 			if (!imp) {
+				R_FREE (sym);
 				break;
 			}
 			imp->name  = strdup (symName);
 			if (!imp->name) {
+				R_FREE (sym);
```
Comment (alvarofe) [inline libr/bin/p/bin_nro.c]: what about goto out_walk_symbol to avoid repeat the same pattern all over?

**Comment by h4ng3r** [comment]
@alvarofe, does it loook better now?

File: libr/bin/p/bin_nro.c [hunk 2]
```diff
@@ -192,6 +192,8 @@ static Sdb *get_sdb(RBinFile *bf) {
 
 static void walkSymbols (RBinFile *bf, RBinNROObj *bin, ut64 symtab, ut64 strtab, ut64 strtab_size, ut64 relplt, ut64 baddr) {
 	int i, import = 0;
+    RBinSymbol *sym;
```
Comment (radare) [inline libr/bin/p/bin_nro.c]: spaces

File: libr/bin/p/bin_nro.c [hunk 3]
```diff
@@ -202,45 +204,47 @@ static void walkSymbols (RBinFile *bf, RBinNROObj *bin, ut64 symtab, ut64 strtab
 		if (!symName) {
 			break;
 		}
-		RBinSymbol *sym = R_NEW0 (RBinSymbol);
+		sym = R_NEW0 (RBinSymbol);
```
Comment (radare) [inline libr/bin/p/bin_nro.c]: why?

File: libr/bin/p/bin_nro.c [hunk 4]
```diff
@@ -250,6 +254,12 @@ static void walkSymbols (RBinFile *bf, RBinNROObj *bin, ut64 symtab, ut64 strtab
 		r_list_append (bin->methods_list, sym);
 		i += 8 - 1;
 	}
+    return;
```
Comment (radare) [inline libr/bin/p/bin_nro.c]: use tabs doode

File: libr/bin/p/bin_nro.c [hunk 5]
```diff
@@ -250,6 +254,12 @@ static void walkSymbols (RBinFile *bf, RBinNROObj *bin, ut64 symtab, ut64 strtab
...
+
+out_walk_symbol:
```
Comment (radare) [inline libr/bin/p/bin_nro.c]: walk outside.. but where? beach?

**Comment by radare** [comment]
https://s-media-cache-ak0.pinimg.com/originals/74/bd/71/74bd71878e28a0f9af544c88b1db0d4d.gif

**Pull request merged**